rtengine/histmatching.cc

```diff
@@ -185,50 +202,48 @@ void RawImageSource::getAutoMatchedToneCurve(std::vector<double> &outCurve)
     
     std::unique_ptr<IImage8> target;
     {
-        int tw = source->getWidth(), th = source->getHeight();
-        float thumb_ratio = float(std::max(tw, th)) / float(std::min(tw, th));
-        float target_ratio = float(std::max(fw, fh)) / float(std::min(fw, fh));
+        RawMetaDataLocation rml;
+        eSensorType sensor_type;
+        double scale;
+        int w = fw / skip, h = fh / skip;
+        std::unique_ptr<Thumbnail> thumb(Thumbnail::loadFromRaw(getFileName(), rml, sensor_type, w, h, 1, false, false));
+        target.reset(thumb->processImage(neutral, sensor_type, fh / skip, TI_Nearest, getMetaData(), scale, false));
```

Comment:
I might have a déjà-vu, or is `Thumbnail::loadFromRaw()` guaranteed to return a `Thumbnail`? If so, please bear with me.

Reply:
this is legit, thanks! I'll fix :+1: